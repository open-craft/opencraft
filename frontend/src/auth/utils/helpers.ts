import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
