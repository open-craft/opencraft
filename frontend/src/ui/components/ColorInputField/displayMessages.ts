export const messages = {
  reset: {
    defaultMessage: 'reset',
    description: ''
  }
};
