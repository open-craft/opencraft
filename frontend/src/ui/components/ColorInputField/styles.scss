
.input-color-picker {
  position: absolute;
  z-index: 2;
}

.reset-value {
  font-size: 14px;
  letter-spacing: 0.14px;
  color: #1a6ebb;
  border: 0;
  padding: 0;
  background-color: transparent;

  &:hover, &:focus {
    text-decoration: underline;
  }
}

.padded {
  margin-left: 7px;
}
