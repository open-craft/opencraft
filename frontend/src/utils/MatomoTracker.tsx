import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
