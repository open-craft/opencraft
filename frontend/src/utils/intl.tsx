import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
