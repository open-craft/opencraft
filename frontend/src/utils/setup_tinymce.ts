import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
