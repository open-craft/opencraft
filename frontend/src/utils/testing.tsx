import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
