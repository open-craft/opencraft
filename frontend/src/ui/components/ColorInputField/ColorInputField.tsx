import * as React from 'react';
// import { useOutsideCallback } from 'global/customHooks';
import {
  FormControl,
  FormGroup,
  FormLabel,
  Row,
  Tooltip,
  OverlayTrigger
} from 'react-bootstrap';

import { SketchPicker } from 'react-color';
import { WrappedMessage } from 'utils/intl';
import { messages as internalMessages } from './displayMessages';
import './styles.scss';

interface ColorInputFieldProps {
  fieldName: string;
  genericFieldName?: string;
  initialValue?: string;
  onChange?: any;
  error?: string;
  messages: any;
  loading?: boolean;
  hideTooltip?: boolean;
}

export const ColorInputField: React.SFC<ColorInputFieldProps> = (
  props: ColorInputFieldProps
) => {
  const pickerContainer = React.useRef<HTMLDivElement>(null);
  const [colorPicker, setColorPicker] = React.useState(false);
  const [selectedColor, setColor] = React.useState(props.initialValue);

  // Get name that can be used with generic displayMessages.
  const genericFieldName = props.genericFieldName || props.fieldName;

  const toggleColorPicker = () => {
    setColorPicker(!colorPicker);
  };

  const resetColor = () => {
    // This will reset the field value to the default theme value
    // TODO: this will need to be updated when the Theme configuration endpoint
    // get updated to improve consistency
    props.onChange(props.fieldName, '');
  };

  const hideColorPickerAndSubmit = () => {
    setColorPicker(false);

    if (selectedColor !== props.initialValue) {
      // Only trigger action if value changed
      props.onChange(props.fieldName, selectedColor);
    }
  };

  /**
   * Bind function to document to detect when
   * user clicks outside color picker.
   */
  const handleClick = (event: any) => {
    if (
      pickerContainer.current &&
      !pickerContainer.current.contains(event.target)
    ) {
      hideColorPickerAndSubmit();
    }
  };
  /**
   * Using effect dependent on color to update bound functions
   * when the color is changed, otherwise it would always pick
   * props.initialColor and never made the request.
   */
  React.useEffect(() => {
    document.addEventListener('mousedown', handleClick);
    return () => {
      document.removeEventListener('mousedown', handleClick);
    };
  }, [selectedColor]);

  const tooltip = !props.hideTooltip ? (
    <Tooltip id="redeployment-status">
      <WrappedMessage messages={props.messages} id={`${props.fieldName}Help`} />
    </Tooltip>
  ) : null;

  const fieldValue = () => {
    if (props.initialValue !== undefined) {
      return props.initialValue;
    }
    return 'Not set';
  };

  return (
    <FormGroup className="color-input">
      <FormLabel>
        <WrappedMessage id={genericFieldName} messages={props.messages} />
      </FormLabel>
      <Row>
        <FormControl
          className="input-field-color"
          name={props.fieldName}
          value={fieldValue()}
          disabled={props.loading}
          onClick={toggleColorPicker}
          readOnly
        />

        <div
          className="input-field-preview"
          style={{
            backgroundColor: props.initialValue
          }}
        />

        {tooltip ? (
          <OverlayTrigger placement="right" overlay={tooltip}>
            <div className="info-icon">
              <i className="fas fa-info-circle" />
            </div>
          </OverlayTrigger>
        ) : (
          <button
            className="reset-value padded"
            type="button"
            onClick={resetColor}
          >
            <WrappedMessage id="reset" messages={internalMessages} />
          </button>
        )}
      </Row>

      {tooltip ? (
        <p>
          <button className="reset-value" type="button" onClick={resetColor}>
            <WrappedMessage id="reset" messages={internalMessages} />
          </button>
        </p>
      ) : null}

      {colorPicker ? (
        <div ref={pickerContainer} className="input-color-picker">
          <SketchPicker
            color={selectedColor}
            onChangeComplete={color => {
              setColor(color.hex);
            }}
          />
        </div>
      ) : null}
    </FormGroup>
  );
};
