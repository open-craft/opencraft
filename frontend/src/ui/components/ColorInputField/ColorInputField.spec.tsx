import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ColorInputField } from './ColorInputField';
import { act } from 'react-test-renderer';

// interface ColorInputFieldProps {
//   fieldName: string;
//   initialValue?: string;
//   onChange?: any;
//   error?: string;
//   messages: any;
//   loading?: boolean;
// }

describe("ColorInputField Component", function() {
  const testMessages = {
    test: {
      defaultMessage: 'Test',
      description: ''
    },
    testHelp: {
      defaultMessage: 'Test helper',
      description: ''
    }
  }

  it('Correctly renders color input field.', () => {
      const tree = setupComponentForTesting(
        <ColorInputField
          fieldName="test"
          messages={testMessages}
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Render with default value.', () => {
      const tree = setupComponentForTesting(
        <ColorInputField
          fieldName="test"
          initialValue="#FFFFFF"
          messages={testMessages}
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Render loading.', () => {
      const tree = setupComponentForTesting(
        <ColorInputField
          fieldName="test"
          initialValue="#FFFFFF"
          messages={testMessages}
          loading={true}
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Render component, check if color picker shows after clicking on field and vanishes after clicking outside.', () => {
      let component = setupComponentForTesting(
        <ColorInputField
          fieldName="test"
          initialValue="#FFFFFF"
          messages={testMessages}
        />
      )

      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      // Click on field button and check if color picker was opened
      act(() => {
        component.root.findByType("input").props.onFocus({preventDefault: () => {}});
      })
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      // Click on field button and check if color picker was opened
      act(() => {
        component.root.findByProps({className: "input-color-picker"}).props.onBlur();
      })
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
  });
});
