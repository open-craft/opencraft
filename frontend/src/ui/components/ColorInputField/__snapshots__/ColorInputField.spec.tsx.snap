// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorInputField Component Correctly renders color input field. 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Test
  </label>
  <div
    className="row"
  >
    <input
      className="input-field-color form-control"
      name="test"
      onFocus={[Function]}
      readOnly={true}
      style={
        Object {
          "color": undefined,
        }
      }
    />
    <div
      className="info-icon"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <i
        className="fas fa-info-circle"
      />
    </div>
  </div>
  <p>
    <button
      className="reset-value"
      onClick={[Function]}
      type="button"
    >
      reset
    </button>
  </p>
</div>
`;

exports[`ColorInputField Component Render component, check if color picker shows after clicking on field and vanishes after clicking outside. 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Test
  </label>
  <div
    className="row"
  >
    <input
      className="input-field-color form-control"
      name="test"
      onFocus={[Function]}
      readOnly={true}
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
      value="#FFFFFF"
    />
    <div
      className="info-icon"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <i
        className="fas fa-info-circle"
      />
    </div>
  </div>
  <p>
    <button
      className="reset-value"
      onClick={[Function]}
      type="button"
    >
      reset
    </button>
  </p>
</div>
`;

exports[`ColorInputField Component Render component, check if color picker shows after clicking on field and vanishes after clicking outside. 2`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Test
  </label>
  <div
    className="row"
  >
    <input
      className="input-field-color form-control"
      name="test"
      onFocus={[Function]}
      readOnly={true}
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
      value="#FFFFFF"
    />
    <div
      className="info-icon"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <i
        className="fas fa-info-circle"
      />
    </div>
  </div>
  <p>
    <button
      className="reset-value"
      onClick={[Function]}
      type="button"
    >
      reset
    </button>
  </p>
  <div
    className="input-color-picker"
    onBlur={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="block-picker "
      style={
        Object {
          "MozBorderRadius": "6px",
          "MozBoxShadow": "0 1px rgba(0,0,0,.1)",
          "OBorderRadius": "6px",
          "OBoxShadow": "0 1px rgba(0,0,0,.1)",
          "WebkitBorderRadius": "6px",
          "WebkitBoxShadow": "0 1px rgba(0,0,0,.1)",
          "background": "#fff",
          "borderRadius": "6px",
          "boxShadow": "0 1px rgba(0,0,0,.1)",
          "msBorderRadius": "6px",
          "msBoxShadow": "0 1px rgba(0,0,0,.1)",
          "position": "relative",
          "width": 170,
        }
      }
    >
      <div
        style={
          Object {
            "borderColor": "transparent transparent #ffffff transparent",
            "borderStyle": "solid",
            "borderWidth": "0 10px 10px 10px",
            "height": "0px",
            "left": "50%",
            "marginLeft": "-10px",
            "position": "absolute",
            "top": "-10px",
            "width": "0px",
          }
        }
      />
      <div
        style={
          Object {
            "MozBorderRadius": "6px 6px 0 0",
            "OBorderRadius": "6px 6px 0 0",
            "WebkitBorderRadius": "6px 6px 0 0",
            "WebkitJustifyContent": "center",
            "alignItems": "center",
            "background": "#ffffff",
            "borderRadius": "6px 6px 0 0",
            "display": "flex",
            "height": "110px",
            "justifyContent": "center",
            "msBorderRadius": "6px 6px 0 0",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "color": "#000",
              "fontSize": "18px",
              "position": "relative",
            }
          }
        >
          #ffffff
        </div>
      </div>
      <div
        style={
          Object {
            "padding": "10px",
          }
        }
      >
        <div
          style={
            Object {
              "marginRight": "-10px",
            }
          }
        >
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#D9E3F0",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#D9E3F0"
            />
          </span>
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#F47373",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#F47373"
            />
          </span>
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#697689",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#697689"
            />
          </span>
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#37D67A",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#37D67A"
            />
          </span>
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#2CCCE4",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#2CCCE4"
            />
          </span>
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#555555",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#555555"
            />
          </span>
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#dce775",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#dce775"
            />
          </span>
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#ff8a65",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#ff8a65"
            />
          </span>
          <span
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "MozBorderRadius": "4px",
                  "OBorderRadius": "4px",
                  "WebkitBorderRadius": "4px",
                  "background": "#ba68c8",
                  "borderRadius": "4px",
                  "cursor": "pointer",
                  "float": "left",
                  "height": "22px",
                  "marginBottom": "10px",
                  "marginRight": "10px",
                  "msBorderRadius": "4px",
                  "outline": "none",
                  "position": "relative",
                  "width": "22px",
                }
              }
              tabIndex={0}
              title="#ba68c8"
            />
          </span>
          <div
            style={
              Object {
                "clear": "both",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <input
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            spellCheck="false"
            style={
              Object {
                "MozBorderRadius": "4px",
                "MozBoxShadow": "inset 0 0 0 1px #ddd",
                "OBorderRadius": "4px",
                "OBoxShadow": "inset 0 0 0 1px #ddd",
                "WebkitBorderRadius": "4px",
                "WebkitBoxShadow": "inset 0 0 0 1px #ddd",
                "border": "0px",
                "borderRadius": "4px",
                "boxShadow": "inset 0 0 0 1px #ddd",
                "boxSizing": "border-box",
                "color": "#666",
                "fontSize": "12px",
                "height": "22px",
                "msBorderRadius": "4px",
                "msBoxShadow": "inset 0 0 0 1px #ddd",
                "outline": "none",
                "padding": "0 7px",
                "width": "100%",
              }
            }
            value="#FFFFFF"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ColorInputField Component Render component, check if color picker shows after clicking on field and vanishes after clicking outside. 3`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Test
  </label>
  <div
    className="row"
  >
    <input
      className="input-field-color form-control"
      name="test"
      onFocus={[Function]}
      readOnly={true}
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
      value="#FFFFFF"
    />
    <div
      className="info-icon"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <i
        className="fas fa-info-circle"
      />
    </div>
  </div>
  <p>
    <button
      className="reset-value"
      onClick={[Function]}
      type="button"
    >
      reset
    </button>
  </p>
</div>
`;

exports[`ColorInputField Component Render loading. 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Test
  </label>
  <div
    className="row"
  >
    <input
      className="input-field-color form-control"
      disabled={true}
      name="test"
      onFocus={[Function]}
      readOnly={true}
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
      value="#FFFFFF"
    />
    <div
      className="info-icon"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <i
        className="fas fa-info-circle"
      />
    </div>
  </div>
  <p>
    <button
      className="reset-value"
      onClick={[Function]}
      type="button"
    >
      reset
    </button>
  </p>
</div>
`;

exports[`ColorInputField Component Render with default value. 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Test
  </label>
  <div
    className="row"
  >
    <input
      className="input-field-color form-control"
      name="test"
      onFocus={[Function]}
      readOnly={true}
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
      value="#FFFFFF"
    />
    <div
      className="info-icon"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <i
        className="fas fa-info-circle"
      />
    </div>
  </div>
  <p>
    <button
      className="reset-value"
      onClick={[Function]}
      type="button"
    >
      reset
    </button>
  </p>
</div>
`;
